### ISSUE TITLE
cmd/compile: rename a bunch of things

### ISSUE THREAD

**Issue by josharian** [golang/go#27167]
Over time, I've noticed many things in the compiler that I'd like to rename, but I haven't because of the merge pain it causes and the review annoyance. I propose to use this issue to accumulate (and bikeshed) a list of renamings that we as a community want. Then a trusted contributor can do a single, automated bulk rename all in one commit.

I think this should happen shortly after the freeze starts. That gives everyone maximum opportunity to get all their CLs submitted. (Doing it early in the cycle is the worst idea, as it potentially breaks three months of private CL work.)

I'll start off the list with:

```
OPROC -> OGO
```

I'd also like new names for all of `Etop`, `Erv`, `Etype`, `Ecall`, `Efnstruct`, `Easgn`, and `Ecomplit`, but don't have good suggestions.

**Comment by randall77** [comment]
OCOM -> OCOMPLIMENT. I always parse the former as COMmunicate.
OMINUS -> ONEGATE. It's not symmetric with OPLUS anymore, but OPLUS is rare.
OIND -> ODEREF.  I'd be happy with a better replacement, but OIND is weak.

There's a lot of stuff in the compiler that could be camel-cased instead of all lowercase.

I want to change the sense of kindNoPointers.  That's a bit more than just a rename, so maybe that's a separate change.

**Comment by josharian** [comment]
This bulk rename went reasonably smoothly for Go 1.12. Let's do it again (assuming we find things we want to rename). Let's use this issue as a scratch pad to jot down things we'd like to rename for Go 1.13, and then near the end of the cycle we'll use a google doc again to discuss and agree on new names.

**Comment by josharian** [comment]
`mayAffectMemory` -> `mightAffectMemory`? (permission vs epistemic uncertainty)

**Comment by josharian** [comment]
For runtime as well as cmd/compile: Consistently use `elem` instead of `value` for map elements.

**Comment by odeke-em** [comment]
Coming here from https://github.com/golang/go/issues/31049#issuecomment-476848887

Perhaps let's rename `disableExporting` to `hideFromExportSym`.

**Comment by mdempsky** [comment]
> Perhaps let's rename disableExporting to hideFromExportSym.

I'm supportive of changes to make that bit of code easier to understand, but my suspicion is this proposed change would have the opposite effect. "disableExport" and its godocs explain the high-level idea of what that function does, whereas "hideFromExportSym" focuses on the low-level details of how it interacts with other export-related functions. Honestly, I can never remember exactly what "exportsym", "autoexport", or a number of other export-related functions do, and always have to look at their code again to remember the details.

I'd suggest a better solution here is just to move "disableExport" into export.go and nearby exportsym, and maybe add an extra internal comment explaining the interaction.

**Comment by cuonglm** [comment]
How about `OPACK` to `OIMPORT` @josharian @mdempsky ?

**Comment by josharian** [comment]
@cuonglm at the end of the cycle I’ll take all these suggestions, put them into a new 1.13 doc, and we can discuss all of them then.

**Comment by cuonglm** [comment]
@josharian I put it in the doc already.

**Comment by josharian** [comment]
@cuonglm I saw that. But I plan to start a new doc; it'll be easier to discuss without having to think about what has already gone in and what hasn't.

**Comment by josharian** [comment]
I missed the window for this for 1.13. Will pick up all the queued renames during the 1.14 freeze.

**Comment by cuonglm** [comment]
@josharian is there a new doc for https://github.com/golang/go/issues/27167#issuecomment-487312858?

**Comment by josharian** [comment]
No, because I didn’t do this for 1.13. There will be one for 1.14. In the meantime, possible additions belong here in this issue.

**Comment by mdempsky** [comment]
Some cmd/compile/internal/types rename suggestions for the next round:

* Drop the "type" suffix from Bytetype, Runetype, and Errortype; just Byte, Rune, Error. It's cleaner.
* Rename Idealfoo to UntypedFoo.
* Rename TIDEAL to TUNTYPED.

**Maybe** suggestions:

* Rename EType and Etype to Kind (and wantEtype to wantKind, etc)?
* Rename TFOO to Foo?
  * To avoid conflicts, we'd probably also need to rename Array, Slice, Map, etc. to ArrayExtra, SliceExtra, MapExtra.

**Comment by mdempsky** [comment]
* Rename OPRINTN to OPRINTLN.

**Comment by josharian** [comment]
@mdempsky oh yes please.

We should probably do these now, since the freeze is going to be a hard stop this cycle.

I don’t suppose you want to send a CL for the outstanding items?
